### ISSUE TITLE
the dialog that is displayed after checking for updates fails to respect Orca's silence speech feature

### ISSUE THREAD

**Issue by ajborka** [microsoft/vscode#93056]
In vscode insiders 1.44, the dialog that is displayed after checking for updates fails to respect Orca's silence speech feature. This makes navigating the dialog difficult at best.
1. Start Orca
2. Start vscode insiders.
3. In the manage toolbar of the main window, press the manage button
4. From the popup menu, choose 'check for updates...'

Actual results: Orca reads the entire contents of the dialog even if the user presses the ctrl key to silence speech. Any actions taken while Orca reads the dialog contents are not spoken until Orca finishes the previous block of text.
Expected result: Interupting Orca's speech should stop Orca from reading the remaining contents of the dialog.

**Comment by isidorn** [comment]
@ajborka is the dialog you are getting "There are currently no updates available"? Can you also reproduce if you do Help > about vs code?
Because I can.

And this smells like we use `aria.alert` for dialogs and it feels like the Orca `ctrl` does not silence `alerts`. @joanmarie is Orca slience designed in such a way?

@deepak1556 also assinging also to you since this is a native dialog and might be somewhat connected to, since if we had nice reading of dialog content we would not have to use `alerts`
https://github.com/electron/electron/issues/14234

**Comment by ajborka** [comment]
@isidorn yes, it happens when getting the no updates available dialog, the help>about dialog, and I suspect the open/save/save as... dialogs might have the same problem, although the lag isn't as bad. Do you want different issues for the open/save/save as.../the 'do you want to save?' dialogs? because they all produce the same problem to some degree.

**Comment by joanmarie** [comment]
Lemme take a look.

**Comment by joanmarie** [comment]
Ok, that's this issue https://bugs.chromium.org/p/chromium/issues/detail?id=1042864

Orca interrupts speech when keys are pressed; not just Ctrl. So you can also see Orca failing to shut up in the About dialog for VSCode as you Tab between its buttons.

The way Orca does this speech interruption is by getting a notification that a key was pressed and then doing the interruption. If Orca isn't told the use pressed a key, it doesn't know it needs to shut up. Sorry! Hopefully one of my colleagues will fix that Chromium issue soon.

**Comment by isidorn** [comment]
@joanmarie Thanks a lot for linking to the chromum issue.

**Comment by isidorn** [comment]
@ajborka can you check if the issue https://bugs.chromium.org/p/chromium/issues/detail?id=1042864 is part of the latest Electron which we have and then maybe we should add a "fixed-in-electron-9" label here?

**Comment by alex19EP** [comment]
> 
> 
> @deepak1556 can you check if the issue https://bugs.chromium.org/p/chromium/issues/detail?id=1042864 is part of the latest Electron which we have and then maybe we should add a "fixed-in-electron-9" label here?
> 
> Also can someone teach me how to figure out what chrome version a fix ends up in? Is that specified somewhere?

i am not @deepak1556 but I can say for sure that the problem is not fixed in electron 9.

as for chromium version, I'd like to know that too. 🙂

**Comment by isidorn** [comment]
Thanks. Then Electron 10 maybe :)

**Comment by deepak1556** [comment]
The fix landed first in `86.0.4199.0`, which would be Electron 11. I will look into backporting to 10 and 9 for next iteration.

**Comment by isidorn** [comment]
Great, thanks. Yeah if the backport is not too complex it would be great to have this.

**Comment by isidorn** [comment]
@deepak1556 we should move this to October?

**Comment by deepak1556** [comment]
Exploration builds based on Electron 11:

[Zip Archive](https://az764295.vo.msecnd.net/exploration/1dfc92ea9b990fad440fcf12ce46cb85897861ce/code-exploration-x64-1603350397.tar.gz)
[Deb](https://az764295.vo.msecnd.net/exploration/1dfc92ea9b990fad440fcf12ce46cb85897861ce/code-exploration_1.51.0-1603349218_amd64.deb)
[Snap](https://az764295.vo.msecnd.net/exploration/1dfc92ea9b990fad440fcf12ce46cb85897861ce/code-exploration-1603350775.snap)
[RPM](https://az764295.vo.msecnd.net/exploration/1dfc92ea9b990fad440fcf12ce46cb85897861ce/code-exploration-1.51.0-1603349307.el7.x86_64.rpm)

**Comment by isidorn** [comment]
Added fixed in electron 11 label, since @jvesouza confirmed this is fixed with exploration builds.

**Comment by webczat** [comment]
finally! I know chromium unstable has this fixed and I admit I was wondering when this would land here. :+1:

**Comment by ajborka** [comment]
@deepak1556 , @isidorn  this appears to be fixed. Nice job.
